packages/nextjs/test/integration/test/server/tracingWithSentryAPI.js

```diff
@@ -0,0 +1,44 @@
+const assert = require('assert');
+
+const { sleep } = require('../utils/common');
+const { getAsync, interceptEventRequest, interceptTracingRequest } = require('../utils/server');
+
+module.exports = async ({ url: urlBase, argv }) => {
+  const urls = {
+    unwrappedNoParamURL: `/api/withSentryAPI/unwrapped/noParams`,
+    wrappedNoParamURL: `/api/withSentryAPI/wrapped/noParams`,
+    unwrappedDynamicURL: `/api/withSentryAPI/unwrapped/dog`,
+    wrappedDynamicURL: `/api/withSentryAPI/wrapped/dog`,
+  };
+
+  const interceptedRequests = {};
+
+  Object.entries(urls).forEach(([testName, url]) => {
+    interceptedRequests[testName] = interceptTracingRequest(
+      {
+        contexts: {
+          trace: {
+            op: 'http.server',
+            status: 'ok',
+            tags: { 'http.status_code': '200' },
+          },
+        },
+        transaction: `GET ${url.replace('dog', '[animal]')}`,
+        type: 'transaction',
+        request: {
+          url: `${urlBase}${url}`,
+        },
+      },
+      argv,
+      testName,
+    );
+  });
+
+  Object.values(urls).forEach(async url => await getAsync(`${urlBase}${url}`));
+
+  await sleep(500);
```

Comment:
I believe the `getAsync` calls aren't awaited because `forEach` doesn't call async functions like a waterfall but just calls them all (basically) at the same time. Because of that and because `sleep` is prone to flake, I suggest we change it to the following:

```suggestion
  await Promise.all(Object.values(urls).map(url => getAsync(`${urlBase}${url}`)));
```

Reply:
Oh, damn, you're totally right. Good catch.

I'll switch it to `Promise.all`, but I think it's a good idea to keep the `sleep` call. In the other tests, we await both `getAsync` and `sleep`. The `Promise.all` takes the place of the former, but it makes sense to me that it might take a moment for nock to register the intercepted requests. So I'm going to keep it, but I'll move it back to the standard `250` input.